src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Symbols
+{
+    [CompilerTrait(CompilerFeature.ModuleInitializers)]
+    public sealed class ModuleInitializersTests : CSharpTestBase
+    {
+        private static readonly CSharpParseOptions s_parseOptions = TestOptions.RegularPreview;
+
+        [Fact]
+        public static void ModuleInitializersNotUsableInCSharp8()
```

Comment:
Consider adding a "happy case" test which verifies there are no diagnostics when using `s_parseOptions`.

Reply:
Is the purpose to verify something about `s_parseOptions` for the rest of the tests or to demonstrate the earliest C# version where `[ModuleInitializer]` is allowed?

Reply:
It would be good to use RegularPreview directly for it, maybe with a comment, so that after the feature ships and we audit the usages of RegularPreview, we decide to keep that one at Regular9 instead of eliminating the parseOptions argument altogether.

Reply:
Also, feel free to add the test in the upcoming diagnostics PR.

Reply:
> so that after the feature ships and we audit the usages of RegularPreview, we decide to keep that one at Regular9 instead of eliminating the parseOptions argument altogether.

We actually should eliminate it and test against default then. So that future versions are tested. We should have one test for 9 though.

---
In reply to: [414929977](https://github.com/dotnet/roslyn/pull/43281#discussion_r414929977) [](ancestors = 414929977)

Reply:
Done.